### PR TITLE
Fix ThinEngine compilation and abstract engine split

### PR DESCRIPTION
**Pull request by sebavan**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15138/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15138/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15138/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15138/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15138/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15138/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/testResults/webgpuplaywright/index.html

**Review by deltakosh — APPROVED** [review]

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15138/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Materials/Textures/Filtering/hdrFiltering.ts [hunk 1]
```diff
@@ -7,10 +7,11 @@ import { Constants } from "../../../Engines/constants";
 import { EffectWrapper, EffectRenderer } from "../../../Materials/effectRenderer";
 import type { Nullable } from "../../../types";
 import type { RenderTargetWrapper } from "../../../Engines/renderTargetWrapper";
+import { Logger } from "../../../Misc/logger";
 
+import "../../../Engines/Extensions/engine.renderTargetCube";
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Textures/Filtering/hdrFiltering.ts:12]: How does it work in a WebGPU only context? This import will inject some methods in `ThinEngine.prototype`, but if we want to only use WebGPU in a ES6 context, we won't have `ThinEngine` defined?

Comment (sebavan) [reply packages/dev/core/src/Materials/Textures/Filtering/hdrFiltering.ts:12]: I do admit that the import in WebGpu will be hard to get so I will work on a fix ASAP

**Pull request merged**